Schemata/sarif-schema.json

```diff
@@ -353,6 +272,54 @@
       }
     },
 
+    "externalFiles": {
```

Comment:
>externalFiles [](start = 5, length = 13)

@lgolding, please review this definition carefully. i know you looked at it in context of the SARIF SDK repo, but in isolation here perhaps other improvements occur to you,

Reply:
Huh. This isn't what I had in mind but I think it's better.

I wasn't going to specify the property names that could appear; I was just going to allow arbitrary property names, and each property value would be an array of `fileLocation`s. And there is text in the change draft that says "that array can have one or more elements if the externalized property is an array, otherwise it can have only one element."

But by explicitly enumerating the property names, you've made those one-element arrays unnecessary.

So yes, this is good, and I'm going to revise my change draft to match.

---
In reply to: [215360196](https://github.com/oasis-tcs/sarif-spec/pull/236#discussion_r215360196) [](ancestors = 215360196)